week3/api-services/property_api.py

```
@@ -0,0 +1,176 @@
+        query_time_ms = (end_time - start_time) * 1000
+        
+        if row:
+            # Convert sqlite3.Row to dict
+            data = dict(row)
+            conn.close()
+            return data, query_time_ms
+        else:
+            conn.close()
+            return None, query_time_ms
+            
+    except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)